xbmc/guilib/Texture.cpp

```diff
@@ -493,3 +507,117 @@ bool CTexture::IsMipmapped() const
 {
   return m_mipmapping;
 }
+
+void CTexture::ConvertToBGRA()
+{
+  if (m_format == XB_FMT_A8R8G8B8)
+    return;
+
+  unsigned int oldFormat = m_format;
+  size_t size = GetPitch() * GetRows();
+  m_format = XB_FMT_A8R8G8B8;
+  size_t newSize = GetPitch() * GetRows();
+
+  unsigned char* oldPixels = m_pixels;
+  unsigned char* newPixels = static_cast<unsigned char*>(KODI::MEMORY::AlignedMalloc(newSize, 32));
+
+  if (newPixels == nullptr)
+  {
+    CLog::Log(LOGERROR, "{} - Could not allocate {} bytes. Out of memory.", __FUNCTION__, newSize);
+    m_format = oldFormat;
```

Comment:
Avoid effects in early returns, as all future early returns would need to duplicate the effect. I would only update internal state on success.

Reply:
This changed with the new commit.